src/identify/index.js

```diff
@@ -118,8 +118,8 @@ function updateSelf (peerSelf, observedAddr) {
   }
 
   for (var i = 0; i < peerSelf.previousObservedAddrs.length; i++) {
-    if (peerSelf.previousObservedAddrs[i].toString() === omh.toString()) {
-      peerSelf.previousObserveredAddrs.splice(i, 1)
+    if (multiaddr(peerSelf.previousObservedAddrs[i]).toString() === omh.toString()) {
```

Comment:
Could you clarify this change? the `previousObserverAddrs is intentionally and array of `multiaddr`, they should not be stored as an array of strings.